experiments/tf_trainer/common/base_model.py

```diff
@@ -22,6 +22,9 @@ class BaseModel(abc.ABC):
 
   def map(self, f: Callable[[tf.estimator.Estimator], tf.estimator.Estimator]
          ) -> 'BaseModel':
+    """
+    Allows models to be extended. e.g. adding preprocessing steps.
```

Comment:
i think to be in line with Google python style you want the first sentence to be directly after the first quotes:
"""Comment here"""

Or maybe:
"""Allows models to be extended.

For example, adding preprocessing steps
"""